### ISSUE TITLE
Instructions: Merge multiple close Continue Instructions

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1282]
Sometimes at more complicated intersections, there might be multiple continues in a very short succession (<50m apart). Creating multiple instructions does not really help the wayfinding, but is rather distracting IMHO.

Here is one [example](https://graphhopper.com/maps/?point=-33.788943%2C151.127254&point=-33.787342%2C151.125891&debug=true). Both continues on it's own are correct. But having them in this case, might not help.

I propose to merge two successive continues if they are close (e.g. < 50m) and if they use the same street name. I have been at this intersection multiple times with the [Kurviger app](https://play.google.com/store/apps/details?id=gr.talent.kurviger.pro) and the two continues make it hard to understand that one has to turn right shortly afterwards.

![gh-continue](https://user-images.githubusercontent.com/1553525/36012391-e9857af4-0db1-11e8-87ec-2ed93e20c358.png)
![gh-continue-2](https://user-images.githubusercontent.com/1553525/36012404-fe58ccf6-0db1-11e8-8c5a-c6115fbfc074.png)

**Comment by jguthrie100** [comment]
Also worth considering for longer gaps on highways and motorways.

Driving along A8 from Munich to Pforzheim has 29 "Continue" or "Keep left" instructions when probably 1 could do (or maybe a few more if the motorway has some strange junctions where the road splits and the "Keep left" is actually required.

**Comment by devemux86** [comment]
> or maybe a few more if the motorway has some strange junctions where the road splits and the "Keep left" is actually required

Indeed where there are junctions or exits in a highway, then the keeps seem to be needed.

**Comment by boldtrn** [comment]
> Also worth considering for longer gaps on highways and motorways.

Yes too many keeps and continues can be annoying. Especially on motorways it's complicated to differentiate between an off-ramp and a forking road. All continues or keeps on the motorway are there, because we are not sure if it's a fork or an off-ramp. The biggest problem is that we cannot store too much information in the graph, because of the bit limit, that should be fixed with #1112. Furthermore this could be improved with #1232.

**Comment by devemux86** [comment]
> Especially on motorways it's complicated to differentiate between an off-ramp and a forking road. All continues or keeps on the motorway are there, because we are not sure if it's a fork or an off-ramp.

That would require also knowledge of (edge) road type during instructions processing.

**Comment by boldtrn** [comment]
> That would require also knowledge of (edge) road type during instructions processing.

Yes, which can be added once #1112 is finished :).

**Comment by karussell** [comment]
I think this is a duplicate of #1007.
